roofit/roofitcore/src/RooNLLVarNew.cxx

```diff
@@ -309,8 +309,17 @@ void RooNLLVarNew::translate(RooFit::Detail::CodeSquashContext &ctx) const
    // brackets of the loop is written at the end of the scopes lifetime.
    {
       auto scope = ctx.beginLoop(this);
-      ctx.addToCodeBody(this, resName + " -= " + ctx.getResult(_weightVar.arg()) + " * std::log(" +
-                                 ctx.getResult(_pdf.arg()) + ");\n");
+      std::string const &weight = ctx.getResult(_weightVar.arg());
+      std::string const &pdfName = ctx.getResult(_pdf.arg());
+
+      if (_binnedL) {
+         // Since we only support uniform binning, bin width is the same for all.
+         std::string muName = pdfName + " * " + std::to_string(_binw[0]);
```

Comment:
```suggestion
         if (!_pdf->getAttribute("BinnedLikelihoodActiveYields")) {
            std::stringstream errorMsg;
            errorMsg << "RooNLLVarNew::translate(): binned likelihood optimization is only supported when raw pdf "
                        "values can be interpreted as yields."
                     << " This is not the case for HistFactory models written with ROOT versions before 6.26.00";
            coutE(InputArguments) << errorMsg.str() << std::endl;
         }
         std::string muName = pdfName;
```
Since #13024, the multiplication with the bin widths is not required anymore, at least for HistFactory models written with ROOT 6.26 or later. I think we don't need to support these older workspaces anymore, and like this we will also not have problems with non-uniform binnings later.